include/ccf/tx_id.h

```diff
@@ -74,7 +74,7 @@ namespace ccf
         const auto seqno_sv = sv.substr(separator_idx + 1);
         const auto [p, ec] =
           std::from_chars(seqno_sv.begin(), seqno_sv.end(), tx_id.seqno);
-        if (ec != std::errc() || p != seqno_sv.end())
+        if (ec != std::errc() || p != seqno_sv.end() || tx_id.seqno < 1)
```

Comment:
```suggestion
        if (ec != std::errc() || p != seqno_sv.end() || tx_id.seqno == SEQNO_UNKNOWN)
```

Semantically identical (since `txid.seqno` is unsigned, any negatives will already not parse), but I think clearer.